### DISCUSSION TITLE
keep getting inconsistent z axis start

### DISCUSSION THREAD

**Discussion by yusufariffin** [Jyers/Marlin#1081] [Q&A] [answered]
using the latest v1.3.4, i keep getting various different z axis first layer height. i have to change my z off set everytime after a print even without switching it off. during my manual levelling, i set my z offset to 0 and saved it right away, when i hit print, the hotend just slam the glass and have to readjust during skirting ranging from -1 to 0.05 each and every print. different height everytime. i swapped the z axis switch to the x axis and still the same problem happen.. i changed my spring bed already.. it is frustrating.. should i flash to the original version of the firmware?

**Comment by olskar** [comment]
I also have this problem, not sure if it's the firmware or me not calibrating it correctly

**Reply by yusufariffin** [reply]
i added BL Touch and now its like having the flattest glass in the world.. couldnt be happier

**Comment by smooreace** [comment, marked as answer]
If your nozzle actually MOVED the bed then you need to relevel.. which
could be why your offset was out of whack.

Level the bed
Home Z
Move Z to 0
adjust Z offset
Save
Add M420 S1 to you start gcode af G28
Prosper

**Reply by yusufariffin** [reply]
thanks for the reply! will try that out, i just bought my first bl touch, gonna flash that bl firmware, tq

**Comment by Ramrawd** [comment]
I'm also having this problem. Z-offset changes after every print when I used to not have to touch it for weeks at a time. I updated to 1.3.4 a few days ago and just noticed this behavior.

**Reply by yusufariffin** [reply]
i upgraded to a BL Touch and its like upgrading the printer, 5 x 5 high speed auto bed levelling is awesome.. no missing spot

**Comment by Reggi3** [comment]
If you're experiencing this issue with 1.3.4, why not roll back to 1.3.3 or earlier and see if you can replicate the issues?  

If you can't then it's possibly something in the firmware (did you wipe the eeprom after updating to 1.3.4?) but if you can replicate the issues, it's likely that it's a hardware issue, possibly involving the bed, with areas of concern being the Y carriage (rollers, eccentric nuts, lock nuts) and also the X carriage (rollers, nuts), X gantry (extrusion not secured properly to the X gantry left bracket).

**Reply by olskar** [reply]
I am not having this problem any longer since downgrading to 1.3.2 at least, not sure what could have changed

**Comment by PossibleCause** [comment]
I also have this unexpected behaviour when starten my ender 3v2 4.2.7 board.

When homing all the z axis stops suddenly at some distance from the bed. I would expect that it only stops when hitting the z-end stop.

**Reply by yusufariffin** [reply]
i upgraded to a BL Touch and its like upgrading the printer, 5 x 5 high speed auto bed levelling is awesome.. no missing spot

**Reply by PossibleCause** [reply]
I discoverd that the cable of the z nd stop was broken. This caused inconsistant operation in homing. After fixing the cable I have no more problems.

**Comment by Gitaryadeller** [comment]
i am having the same problem. z offset stops at different heights at each start. sometimes -1.00, sometimes 0.30, different zofsets appear each time.

**Reply by yusufariffin** [reply]
i upgraded to a BL Touch and its like upgrading the printer, 5 x 5 high speed auto bed levelling is awesome.. no missing spot
